.github/workflows/preview.yml

```diff
@@ -3,43 +3,91 @@ name: Publish Preview release
 on:
   schedule:
     # run every day at 00:15 UTC to avoid high load times at the start of every hour
-    - cron: '15 0 * * *'
+    - cron: "15 0 * * *"
 
 jobs:
-  publish:
-    name: Publish Preview to VS Code Marketplace
-    runs-on: ubuntu-latest
+  build:
+    name: Package
+    strategy:
+      matrix:
+        include:
+          - os: windows-latest
+            platform: win32
+            arch: x64
+            npm_config_arch: x64
+            target: win32-x64
+          - os: windows-latest
+            platform: win32
+            arch: ia32
+            npm_config_arch: ia32
+            target: win32-ia32
+          # - os: windows-latest
+          #   platform: win32
+          #   arch: arm64
+          #   npm_config_arch: arm
+          #   target: win32-arm64
+          - os: ubuntu-latest
+            platform: linux
+            arch: x64
+            npm_config_arch: x64
+            target: linux-x64
+          - os: ubuntu-latest
+            platform: linux
+            arch: arm64
+            npm_config_arch: arm64
+            target: linux-arm64
+          - os: ubuntu-latest
+            platform: linux
+            arch: armhf
+            npm_config_arch: arm
+            target: linux-armhf
+          - os: macos-latest
+            platform: darwin
+            arch: x64
+            npm_config_arch: x64
+            target: darwin-x64
+          - os: macos-latest
+            platform: darwin
+            arch: arm64
+            npm_config_arch: arm64
+            target: darwin-arm64
+    runs-on: ${{ matrix.os }}
     steps:
       - name: Check out repository
         uses: actions/checkout@v2
-
-      - name: Set preview in package.json
-        id: set-preview
-        run: |
-          ./build/preview
-
-      - name: Read extension version
-        id: ext-version
-        run: |
-          content=`cat ./package.json | jq -r .version`
-          echo "::set-output name=content::$content"
-
       - uses: actions/setup-node@v2
         with:
-          node-version-file: '.nvmrc'
-
+          node-version-file: ".nvmrc"
       - name: Install dependencies
         run: npm ci
-
-      - name: vsce package
-        run: npx -- vsce package --out=${{ runner.temp }}/terraform-${{ steps.ext-version.outputs.content }}.vsix
-
+        env:
+          npm_config_arch: ${{ matrix.npm_config_arch }}
```

Comment:
What is `npm_config_arch` doing? I wasn't able to find anything in the `npm ci` docs

Reply:
The [`npm_config`](https://docs.npmjs.com/cli/v7/using-npm/config) part is a special set of environment variables that NPM automatically populates for each `npm run` command, with whatever after the last `_` being accessible as an environment variable.

While I don't think we'll need it for our stuff, I saw this as setting the `arch` used to compile npm dependencies that had platform specific code inside it. So, not our package per se, but packages we depend on that would need to know we're packaging an `arm` build vs ` x64` build.